### ISSUE TITLE
Included edition causes runtime error

### ISSUE THREAD

**Issue by jasonpang** [bevry/editions#19]
My project is targeting node v4.2.6 and my project's third-party dependencies seem to be including the wrong edition of typechecker, extract-opts, requirerefresh, safefs (all of which use editions). Everything compiles fun, but a runtime error is produced with:

bevry/TypeChecker:
```
 ✘ jpang@ubuntu ● node dist/integration-test-server.js
/home/jpang/code/OneSignal-Website-SDK/dist/integration-test-server.js:137042
        export function getObjectType (value /* :mixed */) /* :string */ {
        ^^^^^^

SyntaxError: Unexpected reserved word
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:374:25)
    at Object.Module._extensions..js (module.js:417:10)
    at Module.load (module.js:344:32)
    at Function.Module._load (module.js:301:12)
    at Function.Module.runMain (module.js:442:10)
    at startup (node.js:136:18)
    at node.js:966:3
```

bevry/extract-opts
```
/home/jpang/code/OneSignal-Website-SDK/dist/integration-test-server.js:137518
        module.exports = function (opts, next, config = {}) {
                                                      ^

SyntaxError: Unexpected token =
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:374:25)
    at Object.Module._extensions..js (module.js:417:10)
    at Module.load (module.js:344:32)
    at Function.Module._load (module.js:301:12)
    at Function.Module.runMain (module.js:442:10)
    at startup (node.js:136:18)
    at node.js:966:3
```

```
/home/jpang/code/OneSignal-Website-SDK/dist/integration-test-server.js:137797
                let result, error
                ^^^

SyntaxError: Block-scoped declarations (let, const, function, class) not yet supported outside strict mode
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:374:25)
    at Object.Module._extensions..js (module.js:417:10)
    at Module.load (module.js:344:32)
    at Function.Module._load (module.js:301:12)
    at Function.Module.runMain (module.js:442:10)
    at startup (node.js:136:18)
    at node.js:966:3
```

I've been able to get past these errors by forcing the `es2015` folder to be used instead of the `source` folders, but I'm having trouble understanding what the root cause of the issue is.

I'm using TypeScript 2.0+ nightly with awesome-typescript-loader. Using webpack to compile (latest version v1.13.2).

Any thoughts on what the issue might be?

**Comment by balupton** [comment]
@jasonpang would you be able to setup an example repo that reproduces it with TypeScript and awesome-typescript-loader - so I can test the fix with it?

**Comment by balupton** [comment]
The likely fix for this will be feature detection #7

**Comment by balupton** [comment]
Okay v1.2.0 should fix this
